src/core/coap/coap_secure.cpp

```diff
@@ -115,6 +115,15 @@ otError CoapSecure::SetPsk(const uint8_t *aPsk, uint8_t aPskLength)
     return mDtls.SetPsk(aPsk, aPskLength);
 }
 
+void CoapSecure::SetPsk(const MeshCoP::JoinerPskd &aPskd)
+{
+    static_assert(static_cast<uint16_t>(MeshCoP::JoinerPskd::kMaxLength) <=
+                      static_cast<uint16_t>(MeshCoP::Dtls::kPskMaxLength),
+                  "The maximum length of DTLS PSK is smaller than joiner PSKd");
+
+    IgnoreError(mDtls.SetPsk(reinterpret_cast<const uint8_t *>(aPskd.GetAsCString()), aPskd.GetLength()));
```

Comment:
If we are sure that `mDtls.SetPsk` will success, I think assertion is better at here.